src/main/java/org/opensearch/sdk/ActionListener.java

```diff
@@ -21,13 +21,35 @@
 import java.net.Socket;
 import java.net.UnknownHostException;
 
+/**
+ * A listener for actions on the local port.
+ */
 public class ActionListener {
 
+    /**
+     * Instantiate a new ActionListener.
+     */
+    public ActionListener() {
+        super();
+    }
+
+    /**
+     * Get the local ephemeral port.
+     *
+     * @return The socket address for localhost.
+     * @throws UnknownHostException if the local host name could not be resolved into an address.
+     */
     @SuppressForbidden(reason = "need local ephemeral port")
     protected static InetSocketAddress getLocalEphemeral() throws UnknownHostException {
```

Comment:
Note; I'm not clear why this method is static - I think it should be an instance method which would work better for DI/unit testing scenarios

Reply:
I'm okay with the method itself being `static` although the naming convention of `getX()` normally implies instance methods.  `createLocalEphemeralAddress()` may be better?

Reply:
As it stands I'm not grokking this class, so I'm hesitant to make a new naming recommendation.  To keep things moving I do not think we need to solve that problem in this PR, resolving this comment.